arch/arc/core/mpu/arc_core_mpu.c

```diff
@@ -47,7 +47,7 @@ void configure_mpu_thread(struct k_thread *thread)
 void configure_mpu_stack_guard(struct k_thread *thread)
 {
 #if defined(CONFIG_USERSPACE)
-	if (thread->thread_base.user_options & K_USER) {
+	if ((thread->thread_base.user_options & K_USER) != 0) {
```

Comment:
I wonder if wouldn't be better to have a macro to check for a mask, something like:

#define check_mask(_value, _mask) \
             (_value & _mask) != 0

Reply:
we can have something for this indeed.